.github/workflows/release.yml

```diff
@@ -0,0 +1,75 @@
+name: Create a New Release
+on:
+  workflow_dispatch:
+    inputs:
+      type:
+        description:
+          'The version type of this release. Must be either "feature" or "bug", representing a new feature release or a bug fix release, respectively,'
+        required: true
+
+jobs:
+  release:
+    # Validate the input type before running
+    if: ${{ github.event.inputs.type }} == 'feature' || ${{ github.event.inputs.type }} == 'bug'
+    name: Create a New Release
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://npm.pkg.github.com/
+          cache: npm
+
+      - name: Update Npm
+        run: npm install -g npm@latest
+
+      - name: Set Version (Feature)
+        if: ${{ github.event.inputs.type }} == 'feature'
+        run: |
+          npm version minor
+          cd example
+          npm version minor
+
+      - name: Set Version (Bug Fix)
+        if: ${{ github.event.inputs.type }} == 'bug'
+        run: |
+          npm version patch
+          cd example
+          npm version patch
+
+      # Output the new package version to environment variable
+      - name: Get Package Version
+        run: echo "PKG_VERSION=$(npm pkg get version | tr -d '"')" >> $GITHUB_ENV
+
+      - name: Build OSCAL Viewer
+        working-directory: example
+        run: |
+          npm ci
+          npm run build
+          zip viewer.zip build
+
+      - name: Create a GitHub Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name:
+            ${{ env.PKG_VERSION }}
+          files:
+            example/viewer.zip
+
+      # Commit the new version and push changes to main, as well as a "tag-{version}" branch
+      # And push a new preminor version to develop.
+      # Should only run when triggered for a feature - for bug fixes
+      # this should be done manually since a merge might be needed.
+      - name: Push Changes
+        if: ${{ github.event.inputs.type }} == 'feature'
+        run: |
+          git push origin main
```

Comment:
```suggestion
          git push origin develop:main
```
We are still on the develop branch here. If we want to push to a different branch, main in this case, then we must do the above command. In `git push origin develop:main`, `develop` specifies the source (or current) branch we are on while `main` is the different branch that we want to push to.

Reply:
I don't think we are necessarily pushing from "develop", we are checking out develop, making changes, and then pushing to several remote branches.

Reply:
Looking at this image, it appears the workflow can be run from develop (or any branch for that matter, I just assumed we would require this branch to be run from develop).

![run-publish-workflow](https://user-images.githubusercontent.com/43248318/141538650-89a4b6e3-85e6-439d-ba5b-ecc4fed655c6.PNG)